libs/blocks/quiz-entry/utils.js

```diff
@@ -78,7 +78,10 @@ export const handleSelections = (prevSelections, selectedQuestion, selections) =
   // de-dup any existing data if they use the ml field and cards.
   if (prevSelections.length > 0) {
     prevSelections.forEach((selection) => {
-      if (selection.selectedQuestion === selectedQuestion) {
+      const jsonSelectionSelectedQustion = JSON.stringify(selection.selectedQuestion);
+      const jsonSelectedQuesion = JSON.stringify(selectedQuestion[0].selectedQuestion);
+      const isSameQuestion = jsonSelectionSelectedQustion === jsonSelectedQuesion;
+      if (isSameQuestion) {
```

Comment:
I think we can skip caching here and reduce the lines of code.
```suggestion
  if (JSON.stringify(selection.selectedQuestion) === JSON.stringify(selectedQuestion[0].selectedQuestion)) {
```